src/System.IO.Ports/tests/Legacy/SerialPort/BaudRate.cs

```diff
@@ -3,271 +3,142 @@
     {
-        Console.WriteLine("Verifying 28800 BaudRate");
-
-        if (!VerifyBaudRate(28800))
-        {
-            Console.WriteLine("Err_003!!! Verifying 28800 BaudRate FAILED");
-            return false;
-        }
-
-        return true;
+        Debug.WriteLine("Verifying 28800 BaudRate");
+        VerifyBaudRate(28800);
     }
 
-    public bool BaudRate_1200()
+    [ConditionalFact(nameof(HasNullModem))]
+    public void BaudRate_1200()
     {
-        Console.WriteLine("Verifying 1200 BaudRate");
-
-        if (!VerifyBaudRate(1200))
-        {
-            Console.WriteLine("Err_004!!! Verifying 1200 BaudRate FAILED");
-            return false;
-        }
-
-        return true;
+        Debug.WriteLine("Verifying 1200 BaudRate");
+        VerifyBaudRate(1200);
     }
 
-    public bool BaudRate_115200()
+    [ConditionalFact(nameof(HasNullModem))]
+    public void BaudRate_115200()
     {
-        Console.WriteLine("Verifying 115200 BaudRate");
-
-        if (!VerifyBaudRate(115200))
-        {
-            Console.WriteLine("Err_005!!! Verifying 115200 BaudRate FAILED");
-            return false;
-        }
-
-        return true;
+        Debug.WriteLine("Verifying 115200 BaudRate");
+        VerifyBaudRate(115200);
     }
 
-    public bool BaudRate_MinValue()
+    [ConditionalFact(nameof(HasOneSerialPort))]
+    public void BaudRate_MinValue()
     {
-        Console.WriteLine("Verifying Int32.MinValue BaudRate");
-
-        if (!VerifyException(Int32.MinValue, ThrowAt.Set, typeof(System.ArgumentOutOfRangeException)))
-        {
-            Console.WriteLine("Err_006!!! Verifying Int32.MinValue BaudRate FAILED");
-            return false;
-        }
-
-        return true;
+        Debug.WriteLine("Verifying Int32.MinValue BaudRate");
+        VerifyException(Int32.MinValue, ThrowAt.Set, typeof(ArgumentOutOfRangeException));
     }
 
-    public bool BaudRate_Neg1()
+    [ConditionalFact(nameof(HasOneSerialPort))]
+    public void BaudRate_Neg1()
     {
-        Console.WriteLine("Verifying -1 BaudRate");
-
-        if (!VerifyException(-1, ThrowAt.Set, typeof(System.ArgumentOutOfRangeException)))
-        {
-            Console.WriteLine("Err_007!!! Verifying -1 BaudRate FAILED");
-            return false;
-        }
-
-        return true;
+        Debug.WriteLine("Verifying -1 BaudRate");
+        VerifyException(-1, ThrowAt.Set, typeof(ArgumentOutOfRangeException));
     }
 
-    public bool BaudRate_Zero()
+    [ConditionalFact(nameof(HasOneSerialPort))]
+    public void BaudRate_Zero()
     {
-        Console.WriteLine("Verifying 0 BaudRate");
-
-        if (!VerifyException(0, ThrowAt.Set, typeof(System.ArgumentOutOfRangeException)))
-        {
-            Console.WriteLine("Err_008!!! Verifying 0 BaudRate FAILED");
-            return false;
-        }
-
-        return true;
+        Debug.WriteLine("Verifying 0 BaudRate");
+        VerifyException(0, ThrowAt.Set, typeof(ArgumentOutOfRangeException));
```

Comment:
Yes, there is a horrible repeated pattern using those VerifyException methods which I would love to improve, but I agree that we should pass through once first and then come back to them.